packages/jupyter-ai/src/chat_handler.ts

```diff
@@ -0,0 +1,167 @@
+  /**
+   * Create a new event manager.
+   */
+  constructor(options: ChatHandler.IOptions = {}) {
+    this.serverSettings =
+      options.serverSettings ?? ServerConnection.makeSettings();
+
+    this._poll = new Poll({ factory: () => this._subscribe() });
+    this._poll.start();
+  }
+
+  /**
+   * The server settings used to make API requests.
+   */
+  readonly serverSettings: ServerConnection.ISettings;
+
+  /**
+   * Whether the chat handler is disposed.
+   */
+  get isDisposed(): boolean {
+    return this._isDisposed;
+  }
+
+   /**
+   * Dispose the chat handler.
+   */
+   dispose(): void {
+    if (this.isDisposed) {
+      return;
+    }
+    this._isDisposed = true;
+
+    // Clean up poll.
+    this._poll.dispose();
+
+    this._listeners = []
+
+    // Clean up socket.
+    const socket = this._socket;
+    if (socket) {
+      this._socket = null;
+      socket.onopen = () => undefined;
+      socket.onerror = () => undefined;
+      socket.onmessage = () => undefined;
+      socket.onclose = () => undefined;
+      socket.close();
+    }
+  }
+
+  public addListener(handler: (msg: any) => void): void {
+    this._listeners.push(handler);
+  }
+
+  public removeListener(handler: (msg: any) => void): void {
+    const index = this._listeners.indexOf(handler)
+    if(index > -1) {
+      this._listeners.splice(index, 1)
+    }
+  }
+
+  public sendMessage(msg: any): void {
+    this._socket?.send(msg)
+  }
+
+  public async getHistory(): Promise<any[]> {
+    console.log("Going to get the history messages...")
+    let data: any[] = []
+    try {
+      data = await requestAPI('chats/history', {
+        method: 'GET'
+      });
+    } catch (e) {
+      return Promise.reject(e);
+    }
+    return data;
+  }
+
+  private _onMessage(msg: any): void {
+    this._listeners.forEach(listener => listener(msg));
+  }
+
+  private _subscribe(): Promise<void> {
+    return new Promise<void>((_, reject) => {
+        if (this.isDisposed) {
+            return;
+        }
+        const { token, WebSocket, wsUrl } = this.serverSettings;
+        const url =
+            URLExt.join(wsUrl, CHAT_SERVICE_URL) +
+            (token ? `?token=${encodeURIComponent(token)}` : '');
+        const socket = (this._socket = new WebSocket(url));
+
+        socket.onclose = () => reject(new Error('ChatHandler socket closed'));
+        socket.onmessage = msg => msg.data && this._onMessage(JSON.parse(msg.data));
+    });
+  }
+
+  private _isDisposed = false;
+  private _poll: Poll;
+  private _socket: WebSocket | null = null;
+  private _listeners:  ((msg: any) => void)[] = [];
+}
+
+/**
+ * A namespace for `ChatHandler`.
+ */
+export namespace ChatHandler {
+    /**
+     * The instantiation options for an event manager.
+     */
+    export interface IOptions {
+      /**
+       * The server settings used to make API requests.
+       */
+      serverSettings?: ServerConnection.ISettings;
+    }
+}
```

Comment:
1. The `ChatHandler` class declaration should not include any usage of the `any` type. All types should be explicitly defined under the `ChatHandler` namespace.

2. `ChatHandler.sendMessage()` should accept an object with a defined type; callers should not be required to call `JSON.stringify()` prior to every invocation.